### ISSUE TITLE
``singleton comparison`` - ==/is true message improvement

### ISSUE THREAD

**Issue by orSolocate** [pylint-dev/pylint#8796]
### Current problem

I was wondering about the ``singleton-comparison`` checker I worked on a bit a while ago.
it suggests you to compare singleton using `is` instead of `==` which is fine, for example for `none`.
But why should we use it with `True` and `False`? since comparison returns a Boolean anyway.
 

### Desired solution

For the following code:

`if expr == True` or `if expr is True`

The checker should suggest to modify it to `if expr`.

for false, both `if expr == False` and `if expr is False`  should be modified to `if not expr`.

### Additional context

There is some controversy about  this matter as can be seen in this thread:
https://stackoverflow.com/questions/27276610/boolean-identity-true-vs-is-true

Since it assumes an object has its `__bool__` implemented although I believe that this is most probably the case in any class, since it comes with a default method that returns `True`.
See this thread for some (maybe engineered) corner cases:
https://stackoverflow.com/questions/64913140/is-there-a-way-to-remove-bool-method-from-a-class

**Comment by jacobtylerwalls** [comment]
Do you think we should use inference? Emit a message for `x is True` just if we infer x only takes boolean values?

The "controversy" you point out makes me think this should be a distinct message.

Also, I probably wouldn't want this in the context of an assertion. `assert x is True` seems fine in the context of a regression test: if `x` is unexpectedly `foo`, that should fail. A response on the SO post suggests checking `isinstance(x, bool) and x` for this case, but although that's terrifically pythonic, it strikes me as overly precise to warrant a pylint message when `assert x is True` is practical.

Interestingly, `isinstance()` doesn't even help with the numpy-bool-is-not-a-bool problem:

```python
>>> type(numpy.ndarray([]).all()).mro()
[<class 'numpy.bool_'>, <class 'numpy.generic'>, <class 'object'>]
```

**Comment by orSolocate** [comment]
Ok, so @Pierre-Sassoulas, viewing this thread again, I believe we decide not to pursue this issue, am I right?

**Comment by Pierre-Sassoulas** [comment]
Not necessarily, but we'd need to reuse the logic from the implicit-booleaness checks.
